### DISCUSSION TITLE
Is it known whether the "EasyMeter Q3DB1024 V3.04" is supported?

### DISCUSSION THREAD

**Discussion by fkroepfl** [UtilitechAS/amsreader-firmware#804] [Q&A]
Some of these meters have been installed as sub-meters in our house and the question naturally arises as to whether this meter can be read with the amsreader firmware?

What I have found so far:

- The IR read head on the optical interface D0 provides a serial interface to RS232.
- A telegram supposedly consists of a start bit, 7 data bits, a parity bit and a stop bit. The baud rate is 9,600.
- The data protocol should meet the requirements of DIN EN 62056-21 and DIN EN 62056-61.

Reference documentaion (German only): 
- https://www.easymeter.com/downloads/products/zaehler/Q3D/Easymeter_Q3D_DE_2016-06-15.pdf

**Comment by ArnieO** [comment]
The firmware does handle meters delivering serial pulses, so it could be that it works if you make an adapter that reads the optical port and converts it to electrical pulses. 
You then would need to use the "Pulse" setting:
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/6ae6cc2b-2de2-420d-bfc0-1456ddfcea7b)

**Comment by fkroepfl** [comment]
The EasyMeter obviously has 2 IR interfaces, one with the pulses and a second (D0) with an OBIS-ASCII encoding type. See the Putty log file in the appendix.

The D0 IR interface is of course of interest, as this is the only way to receive the relevant data.

Extract from the Putty (https://www.putty.org/) log file (9600 7E1):

```
/ESY5Q3DB1024 V3.04

1-0:0.0.0*255(1311003434)
1-0:1.8.0*255(00138202.2415163*kWh)
1-0:2.8.0*255(00019552.2463655*kWh)
1-0:21.7.0*255(002498.10*W)
1-0:41.7.0*255(000688.50*W)
1-0:61.7.0*255(000494.82*W)
1-0:1.7.0*255(003681.42*W)
1-0:96.5.5*255(80)
0-0:96.1.255*255(1ESY1311003434)
!
```

- [putty-easymeter.log](https://github.com/user-attachments/files/15810603/putty-easymeter.log)

**Is the data received in this format and structure correctly interpreted and processed by the amsreader firmware?**

**Reply by ArnieO** [reply]
Ping @gskjold

**Reply by gskjold** [reply]
This is DSMR, and as long as it is pushed from the meter without needing to prompt it, it is supported

**Comment by fkroepfl** [comment]
I think for the hardware setup, the Esp8266 ESP-01S with an RS232 adapter would be ideal.

**Hence the question: Can the amsreader firmware be run on this hardware without any problems?**

<img width="372" alt="Bildschirmfoto 2024-06-13 um 15 04 30" src="https://github.com/UtilitechAS/amsreader-firmware/assets/6333880/77f3e5ed-bfb0-4ed5-905a-aa99b113c72e">

**Reply by ArnieO** [reply]
You need a device 4 MB flash. I believe the ESP-01S is only 1 MB?

**Reply by fkroepfl** [reply]
Too bad, because unfortunately I have not yet found another ESP module with such an ideal RS232 adapter 😉.

To clarify: MB stands for MegaByte or MegaBit?

Do you have an idea or a tip for a suitable ESP set with RS232 adapter?

**Reply by ArnieO** [reply]
Mb = Mega bit
MB = Mega Byte

> Do you have an idea or a tip for a suitable ESP set with RS232 adapter?

Not sure what you are looking for here. All ESPs have built-in serial interface, and ESP32's have more than one.
ESP-01 / ESP8266 is now a quite old device that is (to my knowledge) rarely used in new projects.

**Reply by fkroepfl** [reply]
> Mb = Mega bit 
> MB = Mega Byte

Thank you. However, it seems that not all online stores use this information stringently 😉.

**Reply by fkroepfl** [reply]
> Not sure what you are looking for here.

Since my meter is equipped with an RS232, I am looking for a smart hardware solution ESP + RS232 level converter.
There is such a pluggable adapter for the ESP01(s), so I came across this ESP module. That was the only reason.

Data circuits | Control circuits | Voltage
-- | -- | --
0 (space) | Asserted | +3 to +15 V
1 (mark) | Deasserted | −15 to −3 V
